examples/oracles.js

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env node
+
+/*
+ * Copyright 2018 aeternity developers
+ *
+ *  Permission to use, copy, modify, and/or distribute this software for any
+ *  purpose with or without fee is hereby granted, provided that the above
+ *  copyright notice and this permission notice appear in all copies.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
+ *  REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
+ *  AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
+ *  INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
+ *  LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
+ *  OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
+ *  PERFORMANCE OF THIS SOFTWARE.
+ */
+
+'use strict'
+
+
+const AeternityClient = require('../index')
+const WebSocketProvider = require('../lib/providers/ws')
+
+const program = require('commander')
+const axios = require('axios')
+
+let runOracleServer = (options) => {
+
+  console.log(`${options.host} ${options.port} ${options.httpPort}`)
+
+  let client = new AeternityClient(new WebSocketProvider(options.host, options.port))
+
+  let wsProvider = client.provider
+
+  wsProvider.on ('open', function () {
+      console.log ('Websocket connection is open')
+      client.oracles.register ('queryFormat', 'responseFormat', 4, 500, 5)
+    }
+  )
+  wsProvider.on ('registeredOracle', function (oracleId) {
```

Comment:
I think the example should separate the roles of creators and users of oracles. It's difficult to understand the lifecycle from creating to querying and responding if both roles are mixed in the same place.